beckon-mesh/src/main/java/com/technocreatives/beckon/mesh/processor/MessageQueue.kt

```diff
@@ -51,7 +51,7 @@ internal class MessageQueue(
     fun pdus() = pdus.toList()
 
     fun shouldSendMessageImmediately(beckonMessage: BeckonMessage): Boolean {
-        return sendingId == null && (messagesQueue.isEmpty() || shouldGoNext(beckonMessage))
```

Comment:
why though? iirc if the queue is empty, it means there is no previous message, so we can just send.

Reply:
@lenguyenthanh There is a difference between the messagesQueue and the processingMap. What we could check is that the processingMap is empty if we want to short cut it (but shouldGoNext basically checks this). 

The current logic is false, consider the following case:
We've sent a message with responseOpCode 0x01 and are waiting for the reply. Then the messageQueue is empty, processingMap size is 1 waiting for the response. When we send another message with the same responseOpCode the currently impl will shortcut because messageQueue is empty, and not care about that we are already waiting for a response for the very same OpCode already in the processingMap.

This resulted in us getting the wrong ack since we had 2 messages in the processingMap both waiting for the same Ack.

Reply:
I've also added a test now for the MessageQueue to cover this.

Reply:
you're right. Didn't realize that we have tests.